### DISCUSSION TITLE
grammar to read path/to/file.tex

### DISCUSSION THREAD

**Discussion by mnaderhirn** [eclipse-langium/langium#1597] [General]
I would like to define a grammar to read the file and path information of a file
[Playground](https://langium.org/playground?grammar=OYJwhgthYgBAYgSwDYFMDi4owFB9QHYAuIAngiqgFw6ywAUAMmEagB5JqwA%2BsAwmAAOqZMjCdUPWAGUAbsAlSAomADOpVWABmqRbwAKBBZQCUAagDcefSwAWNOvUF2zAXgByAQQCyS2AHIAen9zKxwJAkhqWlgbIlsAflgtSkiIVA8fP38AOn8w5lYOSlhEAh0QVQo0NOi6CKjYdlYCVUQAewJXfyL8vAFhUXESsoqqhvSHasJG5sI2zu6wRBAAEz6cOWMuUdRK6dqpick51o6u-1V5DZV1TR1FXf3jo9TZthaFi9Q1Ug3DbaSJ7jN6TGLHJofebnbqCIx9VggCBlMDIWBeXxUWCBADaAH0wABaABenkJAC0ALo4gA6AHc8ZSAFSBKxAA&content=A4QwLgFg9GD2UDMCWAbApgOjGgHkA)
Could anyone tell me why the example is not working

**Comment by msujew** [comment]
Hey @mnaderhirn,

the parser cannot take outer context into account (see also https://github.com/TypeFox/chevrotain-allstar/issues/1). Since it's still inside of the `Path` rule, it will attempt to parse the `.` character as an `/` which leads to the error you're seeing.

See [a fixed grammar here](https://langium.org/playground?grammar=OYJwhgthYgBAYgSwDYFMDi4owFB9QHYAuIAngiqgFw6ywAUAMmEagB5JqwA%2BsAwmAAOqZMjCdUPWAGUAbsAlSAomADOpVWABmqRbwAKBBZQCUAagDceCQUjVaDQSwAWZgLwA5AIIBZJbAByAHoA81gtSlsIVE9ffwCAOgCrHGZWDkpYRAIdEFUKNCj7Ohs7WHZWAlVEAHsCNwD05LwBYVFxTOzc-NLomhLIsorCarqGsEQQABNmnDljLi7UPILCO37VovK2StH6gNV5WZV1TR1FJZXe4s2hnZHa-dQ1UlnDBclLnsG%2Bh2vt3aPBqCIzNVggCDZMDIWDePxUWBBADaAH0wABaABeXnRAC0ALpIgA6AHcUfiAFRBKxAA&content=A4QwLgFg9GD2UDMCWAbApgOjGgHkA).
